### ISSUE TITLE
Support embedded fonts via @font-face

### ISSUE THREAD

**Issue by RazrFalcon** [RazrFalcon/resvg#541]
Requires WOFF/WOFF2 support first...

**Comment by yisibl** [comment]
We have used [woff2-rs](https://github.com/zimond/woff2-rs/commits/main) in our product, which may not be perfect, but is basically usable.

**Comment by theofficialgman** [comment]
also interested in this as a feature

**Comment by LaurenzV** [comment]
Or should we rather work towards adding support in ttf-parser?

**Comment by RazrFalcon** [comment]
Yes, that's a good start. The only issue with `woff2-rs` is `thiserror`...

Note that we have to rewrite `fontdb` handling first, because `usvg` accepts a non-mutable reference. We would have to populate `fontdb` with embedded fonts and then remove those fonts after `usvg` is finished.

I'm not even sure if we're capable of parsing `@font-face` at all.

If you have any ideas - I'm interested in hearing them.

As always, there is nothing simple when it comes to SVG.

**Comment by LaurenzV** [comment]
> I'm not even sure if we're capable of parsing @font-face at all.

How so?

**Comment by RazrFalcon** [comment]
Because we use `simplecss` and it doesn't support at-rules.

**Comment by maielo** [comment]
+1

**Comment by nicoburns** [comment]
> The only issue with woff2-rs is thiserror...

I have ported https://github.com/google/woff2 (C library used to decode woff in Chrome/Firefox) to Rust as https://github.com/nicoburns/wuff. It supports hmtx decoding, and doesn't depend on `thiserror`.

**Comment by LaurenzV** [comment]
Seems like the only missing piece now is adding support for at-rules in simplecss.
